### ISSUE TITLE
IRC build notifications

### ISSUE THREAD

**Issue by FeodorFitsner** [appveyor/ci#88]
Discussion: http://help.appveyor.com/discussions/suggestions/262-irc-notifications

http://www.nuget.org/packages?q=irc

**Comment by ricardopolo** [comment]
:+1:

**Comment by mquinson** [comment]
:+1:

**Comment by retep998** [comment]
:+1:

**Comment by jayvdb** [comment]
:+1:

**Comment by whitequark** [comment]
:+1:

**Comment by wbraswell** [comment]
This is a CRITICAL missing feature.
What is the ETA to implement this?

**Comment by tierra** [comment]
I really wouldn't call it "critical" since it's perfectly possible to implement a webhook that sends the appropriate IRC notifications for you, in fact, I wouldn't be surprised to find a service out there already that does it, and if there's not, you could even use [GitHub's IRC bot](https://github.com/github/github-services/blob/master/lib/services/irc.rb) code to build your own with minimal effort. It would be really convenient though.

Besides, when is anything in IRC considered "critical" infrastructure anyway? I run an channel on FN with 50+ users at most times, and we've had Travis CI builds running for years now, and we've still never even bothered to setup their IRC notifications still. Email directly to the committer responsible for breaking a build is really the most critical part, otherwise, no-one else really needs to be interrupted from their work with more notifications that aren't relevant to them.

**Comment by pbek** [comment]
I would also love to see this! 👍

**Comment by maek** [comment]
:+1:

**Comment by paulsapps** [comment]
:+1:

**Comment by TheCrimsonDevil** [comment]
:+1:

**Comment by pombr** [comment]
I am going to try out this: https://raw.githubusercontent.com/gridsync/gridsync/def54f8166089b733d166665fdabcad4cdc526d8/misc/irc-notify.py and this https://github.com/gridsync/gridsync/blob/master/appveyor.yml by @crwood to use to solve the IRC part of  https://github.com/nexB/scancode-toolkit/issues/330

**Comment by pombredanne** [comment]
I am going to try out this: https://raw.githubusercontent.com/gridsync/gridsync/def54f8166089b733d166665fdabcad4cdc526d8/misc/irc-notify.py and this https://github.com/gridsync/gridsync/blob/master/appveyor.yml by @crwood to use to solve the IRC part of  https://github.com/nexB/scancode-toolkit/issues/330

**Comment by pombredanne** [comment]
You can see this is in action here:
https://ci.appveyor.com/project/nexB/scancode-toolkit/build/23#L494

The code is here: 
https://github.com/nexB/scancode-toolkit/blob/48aeaf76ce9f53d02223c41c1b2ad1d1ad73b851/etc/scripts/irc-notify.py

The appveyor notification is set in the YAML there: 
https://github.com/nexB/scancode-toolkit/blob/48aeaf76ce9f53d02223c41c1b2ad1d1ad73b851/appveyor.yml#L12

There was also some licensing-related discussion with the original author of the script here: https://github.com/nexB/attributecode/commit/446c1753d6d9ab8b62726051c83a4ef1fe72d0fd#commitcomment-19510350

I will likely simplify the scripts such that it accepts a `--pass` and `--fail` option and comes with a default message format. 

@FeodorFitsner and others: If there is some interest, I can package that in a separate repo with simple setup instructions.

**Comment by paulsapps** [comment]
It would probably be nice if chat.freenode.net port 6667 wasn't hard coded/default, not everyone uses free node :)

**Comment by pombredanne** [comment]
@paulsapps good point.... Pretty dumb of me!

**Comment by FeodorFitsner** [comment]
It will be great, thank you! Though we'd need to use some .NET library for sending to IRC channel. Your example could serve as a reference.

**Comment by pombredanne** [comment]
@FeodorFitsner why .NET would be needed? 
Python is part of all you images, right? (And if you insist on .NET, the same script likely runs as-is on IronPython)

**Comment by FeodorFitsner** [comment]
Yes, it's part of AppVeyor images. I mean "native" integration into AppVeyor UI.

**Comment by Rondom** [comment]
I also guess that a proper feature would need a higher standard in terms of robustness and features, which is best achieved when it is tightly integrated into the AppVeyor code.

The current script is nice and I will likely use it, but using an IRC library might be helpful to deal with error situations, authentication to NickServ, etc.

**Comment by paulsapps** [comment]
Travis-CI IRC notification always create a new connection from the build VM

**Comment by pombredanne** [comment]
@paulsapps and this script takes the same approach. And create a notification, not a message

**Comment by jyrkive** [comment]
@pombredanne We (Battle for Wesnoth developers) have improved the notification script a bit. In particular, our version joins the channel instead of sending an external message, which means that the channel doesn't need to accept external messages.

https://github.com/wesnoth/wesnoth/blob/master/utils/appveyor/irc-notify.py

**Comment by wbraswell** [comment]
@FeodorFitsner 
Has there been any progress on releasing an official IRC notification feature?
Thanks!  :-)

**Comment by Croydon** [comment]
Notifico could help some people in the meantime.

https://github.com/notifico/notifico

**Comment by wbraswell** [comment]
@pombredanne 
Can you please send me a link to the installation & usage instructions for your IRC notification script?
Thanks!

**Comment by mquinson** [comment]
@jyrkive I see you're not using your notification script anymore, and that you removed it from your tree. Could you post it in a gist or something? Thanks,

**Comment by jyrkive** [comment]
@mquinson Here is the final version of our script: https://github.com/wesnoth/wesnoth/blob/b28d8d469af047cbbb20b18757c07b1bfc6afa31/utils/appveyor/irc-notify.py
